source-http-ingest/src/server.rs

```diff
@@ -233,35 +241,67 @@ impl Handler {
         &self,
         collection_path: String,
         request_headers: axum::http::header::HeaderMap,
-        body: serde_json::Value,
-    ) -> anyhow::Result<(StatusCode, Json<serde_json::Value>)> {
+        body: Bytes,
+    ) -> anyhow::Result<(StatusCode, Json<Value>)> {
         if let Some(expected_header) = self.require_auth_header.as_ref() {
             match required_string_header(&request_headers, "Authorization") {
                 Ok(auth) if auth == expected_header => { /* request is authorized */ }
                 Ok(_) => {
                     return Ok(err_response(
                         StatusCode::FORBIDDEN,
                         anyhow::anyhow!("invalid authorization token"),
+                        &collection_path,
                     ));
                 }
                 Err(err) => {
-                    return Ok(err_response(StatusCode::UNAUTHORIZED, err));
+                    return Ok(err_response(
+                        StatusCode::UNAUTHORIZED,
+                        err,
+                        &collection_path,
+                    ));
                 }
             }
         }
 
+        // Try to deserialize the body as json _before_ acquiring the handlers lock,
+        // to prevent slow connection from impacting others.
+        let json: Value = match serde_json::from_slice(&body) {
+            Ok(j) => j,
+            Err(err) => {
+                let content_type = request_headers.get("content-type");
+                let content_encoding = request_headers.get("content-encoding");
+
+                let req_body = String::from_utf8_lossy(&body);
+                tracing::warn!(%collection_path, ?content_type, ?content_encoding, %req_body, error = ?err, "failed parsing request body as json");
+                return Ok(err_response(
+                    StatusCode::BAD_REQUEST,
+                    anyhow::anyhow!("invalid json: {err}"),
+                    &collection_path,
+                ));
+            }
+        };
+
+        if !json.is_object() {
+            return Ok(err_response(
+                StatusCode::BAD_REQUEST,
+                anyhow::anyhow!("request body must be an object"),
+                &collection_path,
+            ));
+        }
+
         let start = std::time::Instant::now();
         tracing::debug!(path = %collection_path, "looking up handler for path");
         let mut handlers_guard = self.handlers_by_path.lock().await;
 
         let Some(collection) = handlers_guard.get_mut(collection_path.as_str()) else {
+            tracing::info!(uri_path = %collection_path, "unknown uri path");
```

Comment:
warn?

Reply:
This was intentional, so that you can turn the log level to `warn` if it gets too spammy due to bots or something. I'm assuming you don't have a very strong opinion here and were just pointing it out because it seemed accidental